opentelemetry-instrumentation/tests/test_auto_tracing.py

```diff
@@ -104,3 +108,24 @@ def test_trace_init_otlp(self):
             "my-otlp-test-service",
         )
         del environ["OTEL_SERVICE_NAME"]
+
+    @patch.dict(
+        environ, {"OTEL_SERVICE_NAME": "my-random-ids-generator-test-service"}
+    )
+    def test_trace_init_default_random_ids_generator(self):
+        Configuration._reset()
+        ids_generator_name = components.get_ids_generator()
+        ids_generator = components.import_ids_generator(ids_generator_name)
+        components.init_tracing({"otlp": OTLPExporter}, ids_generator)
+
+        self.assertEqual(self.set_provider_mock.call_count, 1)
+        provider = self.set_provider_mock.call_args[0][0]
+        self.assertIsInstance(provider, Provider)
+        self.assertIsInstance(provider.ids_generator, RandomIdsGenerator)
+        self.assertIsInstance(provider.processor, Processor)
+        self.assertIsInstance(provider.processor.exporter, OTLPExporter)
```

Comment:
why verify things like the exporter, processor, provider? It's a bit harder to see what this test is trying to verify, when you include assertions for things that aren't related.

I'd argue if you are worried about those, they should be split out into smaller tests. It helps future developers more quickly find what behavior has regressed.

Reply:
I can go either way with this, I think it's useful to know that in a test where "I changed only one thing" the other parts of the functionality here still work.

However cleanliness is nice if I remove the rest of the checks, so I have removed them!